Library/Homebrew/os/mac/pathname.rb

```diff
@@ -1,5 +1,9 @@
 require "os/mac/mach"
 
 class Pathname
-  include MachO
+  if ENV["HOMEBREW_RUBY_MACHO"]
+    include RubyMachO
+  else
+    include CctoolsMachO
+  end
```

Comment:
That makes a lot of sense, fixed!
I also made `os/mac/keg` `require` its library conditionally in a similar fashion.